docs_espressif/en/debugproject.rst

```diff
@@ -1,2 +1,154 @@
+.. _debug your project:
+
 Debug Your Project
 ===============================
+
+Before debbugging the project, the user needs to specify the serial port of the device, select menu **View**, **Command Palette** and type **ESP-IDF: Select Port to Use** command.
+
+Make sure that openOCD configuration files are correct by selecting menu **View**, **Command Palette** the **ESP-IDF: Select OpenOCD Board Configuration** to choose the openOCD configuration files for the extension openOCD server.
+
+.. note::
+  * Please review `Configuration of OpenOCD for Specific Target <https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-guides/jtag-debugging/tips-and-quirks.html#jtag-debugging-tip-openocd-configure-target>`_ to understand which board or configuration to use for your specific hardware.
+  * Make sure to configure your drivers as mentioned in ESP-IDF `Configure JTAG Interface <https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-guides/jtag-debugging/configure-ft2232h-jtag.html>`_ documentation.
+
+To start debugging select menu **Run** and **Start Debugging** or just press F5.
+
+.. image:: ../../media/tutorials/debug/init_halted.png
+
+You can see the output from GDB in the debug console and the openOCD output in the menu **View** -> **Output** -> Select **ESP-IDF** from dropdown output.
+
+This cover the basic functionality of the ESP-IDF extension. Take a look at the :ref:`Additional IDE Features<additional features>` documentation for more.
+
+What happens in the background ?
+-------------------------------------
+
+First OpenOCD server is launched in the background and the output is shown in menu **View** -> **Output** -> Select **ESP-IDF** from the dropdown. By default it will be launched using localhost, port ``4444`` for Telnet communication, port ``6666`` for TCL communication and port ``3333`` for gdb. The user can modify **openocd.tcl.host** and **openocd.tcl.port** configuration settings to modify these values. You can also set **idf.openOcdDebugLevel** to lower or increase (0-4) the messages from OpenOCD in the ESP-IDF output.
+
+Next The `Eclipse CDT GDB Adapter <https://github.com/eclipse-cdt-cloud/cdt-gdb-adapter>`_ is launched in the background and the output is shown in the Debug Console. This debug adapter will start the connection to the device and launch the GDB debug session.
+
+This adapter is a middle man between Visual Studio Code, configured toolchain GDB and OpenOCD server. You can see how `Espressif chips debugging works <https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-guides/jtag-debugging/index.html#how-it-works>`_ and how Visual Studio Code use `Debug adapters <https://microsoft.github.io/debug-adapter-protocol/overview>`_ to communicate with many debug tools.
+
+Using the Visual Studio Code debugger
+----------------------------------------
```

Comment:
Add a Table of Contents (ToC) for all the debugging features offered so that users can easily jump to them.